collect_app/src/test/java/org/odk/collect/android/widgets/UrlWidgetTest.java

```diff
@@ -22,14 +33,74 @@
 @RunWith(RobolectricTestRunner.class)
 public class UrlWidgetTest {
 
+    private CustomTabHelper customTabHelper;
+
+    @Before
+    public void setUp() {
+        customTabHelper = mock(CustomTabHelper.class);
+    }
+
+    @Test
+    public void getAnswer_whenPromptAnswerDoesNotHaveAnswer_returnsNull() {
+        assertThat(createWidget(promptWithAnswer(null)).getAnswer(), nullValue());
+    }
+
     @Test
-    public void usingReadOnlyOptionShouldMakeAllClickableElementsDisabled() {
+    public void getAnswer_whenPromptHasAnswer_returnsAnswer() {
+        UrlWidget widget = createWidget(promptWithAnswer(new StringData("blah")));
+        assertThat(widget.getAnswer().getDisplayText(), equalTo("blah"));
+    }
+
+    @Test
+    public void usingReadOnlyOption_makeAllClickableElementsDisabled() {
         UrlWidget widget = createWidget(promptWithReadOnly());
         Button urlButton = widget.findViewById(R.id.url_button);
         assertThat(urlButton.getVisibility(), equalTo(View.GONE));
     }
 
+    @Test
+    public void clearAnswer_doesNotClearWidgetAnswer() {
+        UrlWidget widget = createWidget(promptWithAnswer(new StringData("blah")));
+        widget.clearAnswer();
+        assertThat(widget.getAnswer().getDisplayText(), equalTo("blah"));
+    }
+
+    @Test
+    public void whenPromptHasAnswer_setsCorrectAnswer() {
```

Comment:
I would rename this `whenPromptHasAnswer_displaysAnswer` and the below one to `whenPromptAnswerDoesNotHaveAnswer_displayEmptyString`.